deployment/common/changeset/set_config_mcms.go

```diff
@@ -0,0 +1,209 @@
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/config"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/gethwrappers"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/mcms"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/timelock"
+	chain_selectors "github.com/smartcontractkit/chain-selectors"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/common/proposalutils"
+	commontypes "github.com/smartcontractkit/chainlink/deployment/common/types"
+)
+
+type ConfigPerRole struct {
+	Proposer  config.Config
+	Canceller config.Config
+	Bypasser  config.Config
+}
+type TimelockConfig struct {
+	MinDelay time.Duration // delay for timelock worker to execute the transfers.
+}
+
+type MCMSConfig struct {
+	ConfigsPerChain map[uint64]ConfigPerRole
+	ProposalConfig  *TimelockConfig
+}
+
+var _ deployment.ChangeSet[MCMSConfig] = SetConfigMCMS
+
+// Validate checks that the MCMSConfig is valid
+func (cfg MCMSConfig) Validate(e deployment.Environment, selectors []uint64) error {
+	if len(cfg.ConfigsPerChain) == 0 {
+		return errors.New("no chain configs provided")
+	}
+	// configs should have at least one chain
+	state, err := MaybeLoadMCMSWithTimelockState(e, selectors)
+	if err != nil {
+		return err
+	}
+	for chainSelector, c := range cfg.ConfigsPerChain {
+		family, err := chain_selectors.GetSelectorFamily(chainSelector)
+		if err != nil {
+			return err
+		}
+		if family != chain_selectors.FamilyEVM {
+			return fmt.Errorf("chain selector: %d is not an ethereum chain", chainSelector)
+		}
+		_, ok := e.Chains[chainSelector]
+		if !ok {
+			return fmt.Errorf("chain selector: %d not found in environment", chainSelector)
+		}
+		_, ok = state[chainSelector]
+		if !ok {
+			return fmt.Errorf("chain selector: %d not found for MCMS state", chainSelector)
+		}
+		if err := c.Proposer.Validate(); err != nil {
+			return err
+		}
+		if err := c.Canceller.Validate(); err != nil {
+			return err
+		}
+		if err := c.Bypasser.Validate(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// setConfigOrTxData executes set config tx or gets the tx data for the MCMS proposal
+func setConfigOrTxData(ctx context.Context, lggr logger.Logger, chain deployment.Chain, cfg config.Config, contract *gethwrappers.ManyChainMultiSig, useMCMS bool) (*types.Transaction, error) {
+	groupQuorums, groupParents, signerAddresses, signerGroups := cfg.ExtractSetConfigInputs()
+	opts := deployment.SimTransactOpts()
+	if !useMCMS {
+		opts = chain.DeployerKey
+	}
+	opts.Context = ctx
+	tx, err := contract.SetConfig(opts, signerAddresses, signerGroups, groupQuorums, groupParents, false)
+	if err != nil {
+		return nil, err
+	}
+	if !useMCMS {
+		_, err = deployment.ConfirmIfNoError(chain, tx, err)
+		if err != nil {
+			return nil, err
```

Comment:
use something like [this ](https://github.com/smartcontractkit/chainlink/blob/fd9d5de76f4d86555472c21fb184ebaa82b143b7/deployment/keystone/deploy.go#L608)or this  https://github.com/smartcontractkit/chainlink/blob/fd9d5de76f4d86555472c21fb184ebaa82b143b7/deployment/helpers.go#L90 to return a meaningful error to the user  

can be done in a quick follow up